### PR TITLE
Prune failing - "'async_generator' object is not iterable"

### PR DESCRIPTION
**Pull request by Sybrand**
- Switch from generator to returning a list (supported in newer python version, but not in this version).
- Fixed retention bug (backups younger than 1 day were not being retained, so the daily rule ended up not working - added retention for "hourly backups" to fix this.)

# Test Links:
[Percentile Calculator](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/)
[MoreCast](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/morecast)
[C-Haines](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/c-haines)
[FireBat](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator)
[FireBat bookmark](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/fire-behaviour-calculator?s=266&f=c5&c=NaN&w=20,s=286&f=c7&c=NaN&w=16,s=1055&f=c7&c=NaN&w=NaN,s=305&f=c7&c=NaN&w=NaN,s=344&f=c5&c=NaN&w=NaN,s=346&f=c7&c=NaN&w=NaN,s=328&f=c7&c=NaN&w=NaN,s=1399&f=c7&c=NaN&w=NaN,s=334&f=c7&c=NaN&w=NaN,s=1082&f=c3&c=NaN&w=NaN,s=388&f=c7&c=NaN&w=NaN,s=309&f=c7&c=NaN&w=16,s=306&f=c7&c=NaN&w=NaN,s=1029&f=c7&c=NaN&w=NaN,s=298&f=c7&c=NaN&w=NaN,s=1108&f=c5&c=NaN&w=NaN,s=836&f=c7&c=NaN&w=NaN#state=2ec784ca-c46a-49d0-b2b3-1cf32a9015a2&session_state=7d9447c8-db66-4661-b4cb-03d2ac0d1d8f&code=32292df4-2bdf-4f90-a4a8-c8dbcda682a9.7d9447c8-db66-4661-b4cb-03d2ac0d1d8f.2b63f390-f3dc-43ae-89f2-016453863476)
[Fire Behaviour Advisory](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/fire-behaviour-advisory)
[HFI Calculator](https://wps-pr-1487.apps.silver.devops.gov.bc.ca/hfi-calculator)

### CODE REVIEW

**Review by conbrad — APPROVED** [review]

File: openshift/s3-backup/docker/README.md [hunk 1]
```diff
@@ -42,3 +42,15 @@ brew reinstall zlib bzip2
 poetry env use /Users/[user]/.pyenv/versions/3.6.8/bin/python
 poetry install
 ```
+
+### Ubunut specific instructions
```
Comment (andrea-williams) [inline openshift/s3-backup/docker/README.md:46]:
```suggestion
### Ubuntu specific instructions
```

File: openshift/s3-backup/docker/README.md [hunk 2]
```diff
@@ -42,3 +42,15 @@ brew reinstall zlib bzip2
...
+
+If you have trouble getting poetry to use python 3.6.*
```
Comment (andrea-williams) [inline openshift/s3-backup/docker/README.md:48]: I thought we'd upgraded to a new version of python? Like 3.8 or something?

File: openshift/s3-backup/docker/prune_test.py [hunk 3]
```diff
@@ -38,30 +62,53 @@ def __init__(self, methodName: str = ...) -> None:
                           'backup/blah_wps/2021/10/blah_wps_2019-10-20_20-00-16.sql.gz'  # 22
                           ]
 
+    def test_delete_sample_set(self):
+        """ Test what should be deleted using an asecending list of files """
```
Comment (andrea-williams) [inline openshift/s3-backup/docker/prune_test.py:66]:
```suggestion
        """ Test what should be deleted using an ascending list of files """
```

File: openshift/s3-backup/docker/prune_test.py [hunk 4]
```diff
@@ -38,30 +62,53 @@ def __init__(self, methodName: str = ...) -> None:
...
+        files_to_delete = list(decide_files_to_delete(self.ascending_list.copy()))
+        files_to_delete.sort()
+        self.assertNotIn(self.ascending_list[-1], files_to_delete)
+        self.assertNotIn(self.ascending_list[-2], files_to_delete)
+        self.assertNotIn(self.ascending_list[-3], files_to_delete)
+        self.assertNotIn(self.ascending_list[-4], files_to_delete)
+        self.assertNotIn(self.ascending_list[-5], files_to_delete)
+
+    def test_keep_sample_set(self):
+        """ Test what shold be kept using an ascending list of files """
```
Comment (andrea-williams) [inline openshift/s3-backup/docker/prune_test.py:76]:
```suggestion
        """ Test what should be kept using an ascending list of files """
```

File: openshift/s3-backup/docker/prune_test.py [hunk 5]
```diff
@@ -38,30 +62,53 @@ def __init__(self, methodName: str = ...) -> None:
...
+        files_to_keep.sort()
+        self.assertIn(self.ascending_list[-1], files_to_keep)
+        self.assertIn(self.ascending_list[-2], files_to_keep)
+        self.assertIn(self.ascending_list[-3], files_to_keep)
+        self.assertIn(self.ascending_list[-4], files_to_keep)
+        self.assertIn(self.ascending_list[-5], files_to_keep)
+
     def test_keep_small_set(self):
-        """ We only care about the most recent backup if there's nothing older than a day """
+        """ If there's nothing old than a day, we expect to retain up to 5 "hourly" values """
```
Comment (andrea-williams) [inline openshift/s3-backup/docker/prune_test.py:86]:
```suggestion
        """ If there's nothing older than a day, we expect to retain up to 5 "hourly" values """
```

Comment (andrea-williams) [reply openshift/s3-backup/docker/prune_test.py:86]: Poor Sybrand... I know you tried...

**Review by andrea-williams — APPROVED** [review]
Technically Cypress tests are failing, but I can't imagine that your file changes are actually the cause.

File: openshift/s3-backup/docker/README.md (hunk 2 above)
Comment (Sybrand) [reply openshift/s3-backup/docker/README.md:48]: For our API and Ubuntu - yes. For the backup image - no.

**Pull request merged**